darshan-util/pydarshan/tests/test_plot_exp_common.py

```diff
@@ -0,0 +1,176 @@
+            "examples/example-logs/sample-badost.darshan",
+            "POSIX",
+            plot_access_histogram,
+            [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 131072, 0, 0, 0, 0],
+        ),
+        # subtle case where "POSIX" and "MPIIO"
+        # modules show different results
+        (
+            "tests/input/sample-dxt-simple.darshan",
+            "POSIX",
+            plot_access_histogram,
+            [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0],
+        ),
+        (
+            "tests/input/sample-dxt-simple.darshan",
+            "MPI-IO",
+            plot_access_histogram,
+            [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0],
+        ),
+        # more obvious case where "POSIX" and
+        # "MPIIO" modules show different results
+        pytest.param(
+            "imbalanced-io.darshan",
+            "POSIX",
+            plot_access_histogram,
+            [18, 2492, 14679, 0, 50486, 186, 0, 0, 0,
+            0, 43, 301, 2, 0, 50486, 0, 0, 0, 0, 0],
+            marks=pytest.mark.skipif(not pytest.has_log_repo, # type: ignore
+                                    reason="missing darshan_logs"),
+        ),
+        pytest.param(
+            "imbalanced-io.darshan",
+            "MPI-IO",
+            plot_access_histogram,
+            [11, 2492, 2, 0, 0, 0, 0, 410, 86, 0, 2526,
+            303, 2, 0, 97812, 396, 0, 410, 86, 0],
+            marks=pytest.mark.skipif(not pytest.has_log_repo, # type: ignore
+                                    reason="missing darshan_logs"),
+        ),
+        (
+            "examples/example-logs/dxt.darshan",
+            None,
+            plot_opcounts,
+            [6126, 1497, 264, 1379, 5597, 0, 0, 0, 0, 0, 0, 0,
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 39, 0, 1, 0, 0, 0, 0],
+        ),
+        (
+            "examples/example-logs/ior_hdf5_example.darshan",
+            None,
+            plot_opcounts,
+            [36, 23, 22, 4, 53, 0, 0, 36, 23, 1, 0, 0, 0,
+            0, 0, 0, 16, 0, 0, 0, 0, 0, 128, 1, 0, 0, 0, 9],
+        ),
+        (
+            "examples/example-logs/sample-badost.darshan",
+            None,
+            plot_opcounts,
+            [0, 131072, 2048, 2048, 131072, 0, 2048, 0, 0, 0, 0, 0, 0,
+            0, 0, 0, 0, 0, 0, 0, 0, 34816, 97, 6144, 0, 0, 0, 2056],
+        ),
+        (
+            "examples/example-logs/shane_macsio_id29959_5-22-32552-7035573431850780836_1590156158.darshan",
+            None,
+            plot_opcounts,
+            [6, 7816, 51, 32, 4, 0, 0, 0, 7695, 0, 0, 0, 0,
+            0, 0, 64, 16, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+        ),
+    ],
+)
+def test_bar_heights(filename, mod, fig_func, expected_heights, select_log_repo_file):
+    # check bar graph heights
+
+    logs_repo_path = select_log_repo_file
+    if logs_repo_path is not None:
+        # for logs repo cases
+        log_path = select_log_repo_file
+    else:
+        log_path = filename
```

Comment:
Isn't this logic unnecessarily complex? My rough draft was mostly intended to inspire something much better.

```diff
--- a/darshan-util/pydarshan/tests/test_plot_exp_common.py
+++ b/darshan-util/pydarshan/tests/test_plot_exp_common.py
@@ -152,13 +152,7 @@ def test_xticks_and_labels(log_path, func, expected_xticks, expected_xticklabels
 )
 def test_bar_heights(filename, mod, fig_func, expected_heights, select_log_repo_file):
     # check bar graph heights
-
-    logs_repo_path = select_log_repo_file
-    if logs_repo_path is not None:
-        # for logs repo cases
-        log_path = select_log_repo_file
-    else:
-        log_path = filename
+    log_path = select_log_repo_file or filename
 
     report = darshan.DarshanReport(log_path)
     fig, ax = plt.subplots()

```

Reply:
Yes, this is much simpler so I've made this adjustment with the latest commit.